sympy/integrals/transforms.py

```diff
@@ -1931,7 +1997,14 @@ def laplace_transform(f, t, s, legacy_matrix=True, **hints):
             else:
                 return type(f)(*f.shape, elements_trans)
 
-    return LaplaceTransform(f, t, s).doit(**hints)
+    if isinstance(f, list):
+        return [laplace_transform(x, t, s, diff_subs=diff_subs, **hints)
+                for x in f]
+
+    LT = LaplaceTransform(f, t, s).doit(**hints)
+    for _f, _r in diff_subs.items():
+        LT = laplace_transform_replace_ic(LT, _f, _r[0], t, s, _r[1])
```

Comment:
Would it not be better to handle this directly in `_laplace_rule_diff`?

Reply:
I thought about this one, and the answer is no. The rules themselves should just do whatever is correct in an abstrtact manner, filling in additional information like actual names of transforms and initial conditions has got nothing to do with the rule itself.